experiment/simulation/js/main.js

```diff
@@ -1 +1,711 @@
+			this.height = height;
+			this.angle = angle;
+			this.pos = [x, y];
+			this.grooveDims = [0, 0];
+		};
+
+		draw(ctx) {
+			ctx.fillStyle = "#b86d29";
+			const e1 = [this.pos[0] + this.width, this.pos[1] + this.height], e2 = [...this.pos], gradX = (e1[0] - e2[0]) / 2, gradY = (e1[1] - e2[1]) / 4;
+			const inter = [((10 - this.grooveDims[0]) * e1[0] + this.grooveDims[0] * e2[0]) / 10, ((10 - this.grooveDims[0]) * e1[1] + this.grooveDims[0] * e2[1]) / 10];
+
+			ctx.beginPath();
+			ctx.moveTo(e2[0], e2[1]);
+			ctx.bezierCurveTo(e2[0] - 3 * gradX, e2[1] + gradY, e1[0] - 2.5 * gradX, e1[1] + gradY, e1[0], e1[1]);
+			ctx.lineTo(e1[0] - this.grooveDims[1] / 2, e1[1]);
+			ctx.lineTo(inter[0] - this.grooveDims[1] / 2, inter[1]);
+			ctx.lineTo(inter[0] + this.grooveDims[1] / 2, inter[1]);
+			ctx.lineTo(e2[0], e2[1]);
+
+			ctx.moveTo(e1[0], e1[1]);
+			ctx.bezierCurveTo(e1[0] + 3.5 * gradX, e1[1] - gradY, e2[0] + 2 * gradX, e2[1] - gradY, e2[0], e2[1]);
+			ctx.lineTo(inter[0], inter[1]);
+			ctx.lineTo(inter[0] + this.grooveDims[1] / 2, inter[1]);
+			ctx.lineTo(e1[0] + this.grooveDims[1] / 2, e1[1]);
+
+			ctx.closePath();
+			ctx.fill();
+		};
+
+		groove(change) {
+			this.grooveDims[0] += change;
+
+			if(this.grooveDims[0] >= 10)
+			{
+				return 1;
+			}
+
+			return 0;
+		};
+	};
+
+	class water {
+		constructor(height, width, x, y) {
+			this.height = height;
+			this.width = width;
+			this.pos = [x, y];
+		};
+
+		draw(ctx) {
+			ctx.fillStyle = "#1ca3ec";
+
+			ctx.beginPath();
+			ctx.rect(this.pos[0], this.pos[1], this.width, this.height);
+			ctx.closePath();
+			ctx.fill();
+		};
+
+		mixing(unit) {
+			this.height -= unit;
+			this.pos[1] += unit;
+		};
+	};
+
+	class casagrande {
+		constructor(height, width, x, y) {
+			this.height = height;
+			this.width = width;
+			this.pos = [x, y];
+			this.img = new Image();
+			this.img.src = './images/casagrande.png';
+			this.img.onload = () => { ctx.drawImage(this.img, this.pos[0], this.pos[1], this.width, this.height); }; 
+		};
+
+		draw(ctx) {
+			ctx.drawImage(this.img, this.pos[0], this.pos[1], this.width, this.height);
+		};
+	};
+
+	class casagrandeSide {
+		constructor(height, width, angle, x, y) {
+			this.height = height;
+			this.width = width; 
+			this.angle = angle; 
+			this.pos = [x, y];
+			this.knobAngle = angle;
+		};
+
+		draw(ctx) {
+			// Lower horizontal part
+			const vertDivide = 0.60;
+			ctx.fillStyle = "#82663c";
+			ctx.lineWidth = lineWidth;
+			ctx.beginPath();
+			ctx.rect(this.pos[0], this.pos[1] + vertDivide * this.height, this.width, (1 - vertDivide) * this.height);
+			ctx.closePath();
+			ctx.fill();
+			ctx.stroke();
+
+			// Right vertical part
+			const horizDivide = 0.80;
+			ctx.beginPath();
+			ctx.rect(this.pos[0] + horizDivide * this.width, this.pos[1], (1 - horizDivide) * this.width, vertDivide * this.height);
+			ctx.closePath();
+			ctx.fill();
+			ctx.stroke();
+
+			// Main cup part
+			ctx.fillStyle = "#dbfe5b";
```

Comment:
Can we create a mnemonic const at the top of this file with this color code? So we have some idea about which color is used here.
Same comment for all the hard-coded color codes